src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/tree/TreeNode.java

```diff
@@ -0,0 +1,125 @@
+package de.terrestris.shogun2.model.tree;
+
+import javax.persistence.Entity;
+import javax.persistence.Inheritance;
+import javax.persistence.InheritanceType;
+import javax.persistence.Table;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+
+import de.terrestris.shogun2.model.PersistentObject;
+
+/**
+ * The TreeNode is an abstract class which is the granular part of an tree.
+ *
+ * @author Kai Volland
+ * @author Daniel Koch
+ *
+ */
+@Entity
+@Table
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
+public abstract class TreeNode extends PersistentObject{
```

Comment:
@dnlkoch and I liked the idea of having all nodes in one table. No matter if it is a folder or a leaf.